CODE_REVIEW.md

```diff
@@ -0,0 +1,20 @@
+## Problems or Code Smells in App
+
+1. While using 'ngFor' use 'trackBy' function to add unique identifier for each item. So that when a particular element changes Angular will know and make DOM chnages only for that element.
+2. In Storage.service.ts, error handling is not implemented in readFromStorage() function while accessing an item in localstorage. Error should be caught and return meaning full message to user or initialize state accordingly. And also while setting item in local storage need to add error handling to catch storage full errors and display meaning full message to user
+3. Publisher name for 'Eloquent JavaScript' book is displayed empty in search list. This gives poor user experience empty cases to be handled to display explanatory messages like 'Publisher not mentioned'.
+4. Book description is displyed using innerHtml. Innerhtml usage could lead to Cross site scripting attacks and there is no proper validation of content provided, so broken html can be passed in which lead to security and unexpected problems.
+5. In total-count component unused ngOnInit() hook is defined that caused uneccesary implementation of OnInit interface. It is good practise to get rid of unused code causing uneccesary implementing of interface.
+6. In readinglist and totalcount components readingList$  and totalUnread$ are not declared as observables. Always declare type for variables in typescript
```

Comment:
This is not entirely true. Sometimes a type can be instantly inferred, so there is no need to specifically specify it, take a look at [this discussion](https://stackoverflow.com/questions/46799828/tslint-type-trivially-inferred-why-is-it-bad-practice-to-include-the-type-he) and [this documentation](https://palantir.github.io/tslint/rules/no-inferrable-types/)